core/src/test/scala/integration/kafka/api/PlaintextProducerSendTest.scala

```diff
@@ -120,17 +121,17 @@ class PlaintextProducerSendTest extends BaseProducerSendTest {
     }
   }
 
-  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testSendWithInvalidCreateTime(quorum: String): Unit = {
+  @ParameterizedTest
+  @MethodSource(Array("timestampConfigProvider"))
+  def testSendWithInvalidBeforeAndAfterTimestamp(messageTimeStampConfig: String, recordTimestamp: Long): Unit = {
```

Comment:
please modify the "valid" test as well to add the usage of this new configuration which should test for parameters where value is equal to timestamp etc.

Reply:
added more tests cases for valid scenarios of 1) when topic config threshold is the same us the record timestamp 2) when record timestamp is within range of threshold